Fix for .navbar-toggleable-* > .container-fluid

zalog opened the pull request:
Problems:
- childrens inside `.navbar-toggleable-* > .container-fluid` are not staying on the same line. You can see this on bigger viewports than `*`.
- when `.container|.container-fluid > .navbar-toggler-*` left/right position is relative to `.container` and not `.navbar` itself. That means that the positioning will be doubled by the padding of the `.navbar`.

We need to set `display: flex` on `.container-fluid` and reset the `.container|.container-fluid > .navbar-toggler-*` position.

Please look at this example: http://codepen.io/zalog/pen/PWRvrJ

Later edit: In my example I've used `.container-fluid.w-100` and I think we need to set `.container-fluid` 100% width by default. Please let me know if so.

zalog committed:
Fix for .navbar-toggleable-* > .container-fluid

nextgenthemes commented:
This partly fixes https://github.com/twbs/bootstrap/issues/21530#issuecomment-280272925

zalog committed:
.container-fluid media-breakpoint-down padding

zalog committed:
positioning .container > .navbar-toggler-*

`.navbar-toggler-*` needs to be left/right: 0 when it's in a `.container` or `.container-fluid` because of position: relative of their parents.

zalog committed:
remove whitespace

zalog commented:
I made some small changes and I updated the description too.

nextgenthemes commented:
If this is the right way to do things, still feels weird to me but I am really new to flexbox, then I think this should have the 100% width come with this patch because that is what people expect. That w-100 class should not be needed IMO.

zalog commented:
Yep, I said so in my description. But for now, let's wait a bit :)
